### DISCUSSION TITLE
Bug Report - Large Engine Damaged state sprite depth does not match the editor setting

### DISCUSSION THREAD

**Discussion by ChaosRifle** [FakeFishGames/Barotrauma#14965] [Bug reports]
### Disclaimers

- [x] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Setting a large engine to have a modified sprite depth, ie, 0.900 instead of default 0.800, has the damaged state of the engine when broken down use what appears to be the default 0.800 instead of the specified sprite depth, resulting in the engine changing sprite depth when damaged.

This is an issue where a room adjacent to the engine has part of the sprite inside it, so the sprite depth is changed to contain the visual of the engine to a single room.

### Reproduction steps

1. Open sub editor
2. load a sub or set a sub up with a large engine
3. set large engine to sprite depth 0.900
4. add additional items on top of the large engine at sprite depth 0.850
5. test the mission in a way such to result in the engine breaking down and changing its sprite to the damaged state. IE powering and allowing to degrade when wet, or firing a grenade launcher at it until it is broken.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.17.0 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Linux

**Comment by jigardi** [comment]
Hmmm... I think I've had this happen while working on a mod the overrides all engines.
Thanks for sharing!
